src/webgpu/shader/execution/flow_control/for.spec.ts

```diff
@@ -0,0 +1,68 @@
+export const description = `
+Flow control tests for for-loops.
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../gpu_test.js';
+
+import { runFlowControlTest } from './harness.js';
+
+export const g = makeTestGroup(GPUTest);
+
+g.test('for_basic')
+  .desc('Test that flow control executes a for-loop body the correct number of times')
+  .fn(t => {
+    runFlowControlTest(
+      t,
+      f =>
+        `
+  ${f.expect_sequence(0)}
+  for (var i = ${f.value(0)}; i < ${f.value(3)}; i++) {
```

Comment:
Why not just `for (var i = 0; i < 3; i++)` are you worried about loop unrolling here? Const-eval shouldn't matter.

Reply:
Yes, loop unrolling can mess with things in ways I'd like to avoid.
Might be good to parameterize these tests to allow the constants to be embedded as literals. Best of both worlds then.